tests: arm: sw_vector_table: enhance test configs for frdm_mcxa166

hakehuang opened the pull request:
irq 0 in frdm_mcxa166/276 are reserved can not used for testing. add a config to shift test irq, also add support for rt700

fixes: #92877, #92521

hakehuang commented:
tests on frdm_mcxa166/frdm_mcxa276/mimxrt700_evk_cm33/mimxrt1180_evk_cm33
```
*** Booting Zephyr OS build v4.2.0-rc2-49-g732a3a5c6655 ***
Running TESTSUITE vector_table
===================================================================
START - test_arm_irq_vector_table
Test *** Booting Zephyr OS build v4.2.0-rc1-249-g2d939cac9283 ***
Running TESTSUITE vector_table
===================================================================
START - test_arm_irq_vector_table
Test Cortex-M IRQs installed directly in the vector table
isr0 ran!
isr1 ran!
isr2 ran!
 PASS - test_arm_irq_vector_table in 0.009 seconds
===================================================================
TESTSUITE vector_table succeeded

------ TESTSUITE SUMMARY START ------

SUITE PASS - 100.00% [vector_table]: pass = 1, fail = 0, skip = 0, total = 1 duration = 0.009 seconds
 - PASS - [vector_table.test_arm_irq_vector_table] duration = 0.009 seconds

------ TESTSUITE SUMMARY END ------

===================================================================
PROJECT EXECUTION SUCCESSFUL

```

wearyzen commented on tests/arch/arm/arm_irq_vector_table/src/arm_irq_vector_table.c:
```diff
@@ -13,7 +14,11 @@
+#ifndef CONFIG_ISR_OFFSET
 #define _ISR_OFFSET 0
+#else
+#define _ISR_OFFSET CONFIG_ISR_OFFSET
```
nit: its juts 3 boards so why not follow the [same approach](https://github.com/zephyrproject-rtos/zephyr/pull/92881/files#diff-e98d660a1f2482bf64b3af6ead5b2f77e0842498f12eeb906d3164c16dc3d9a4L17) as NORDIC?

hakehuang replied:
@wearyzen you mean. 
https://github.com/zephyrproject-rtos/zephyr/blob/ce25bad2a5d491656986bd285a7adaf4db985da6/tests/arch/arm/arm_irq_vector_table/src/arm_irq_vector_table.c#L154
there a two concerns for this:
1. NXP irq setting is not aligned over different soc series. there will be more and more variable in macros, which use a board config would be more clean.
2. I guess this would not be only NXP platform dedicated, other platform can apply the same logic without need to change the test source code.

hakehuang committed:
tests: arm: sw_vector_table: enhance test configs for frdm_mcxa166

irq 0 in frdm_mcxa166/276 are reserved can not used for testing.
add a config to shift test irq, also add support for rt700
also fix irq1 issue on mimxrt1180 as it is a debug trace interrupt

fixes: #92877, #92521

Signed-off-by: Hake Huang <hake.huang@nxp.com>

wearyzen commented on tests/arch/arm/arm_irq_vector_table/src/arm_irq_vector_table.c:
```diff
@@ -13,7 +14,11 @@
  * Offset (starting from the beginning of the vector table)
  * of the location where the ISRs will be manually installed.
  */
+#ifndef CONFIG_ISR_OFFSET
```
CONFIG_ISR_OFFSET will always be defined so do we still need this?
